Automattic-Tracks-iOS/ABTesting/ExPlat.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+import UIKit
+
+@objc public class ExPlat: NSObject, ABTesting {
+    public static var shared: ExPlat!
+
+    private let service: ExPlatService
+
+    private let assignmentsKey = "ab-testing-assignments"
+    private let ttlDateKey = "ab-testing-ttl-date"
+
+    private var ttl: TimeInterval {
+        guard let ttlDate = UserDefaults.standard.object(forKey: ttlDateKey) as? Date else {
+            return 0
+        }
+
+        return ttlDate.timeIntervalSinceReferenceDate - Date().timeIntervalSinceReferenceDate
+    }
+
+    private(set) var scheduledTimer: Timer?
+
+    public init(configuration: ExPlatConfiguration,
+         service: ExPlatService? = nil) {
+        self.service = service ?? ExPlatService(configuration: configuration)
+        super.init()
+        subscribeToNotifications()
+        ExPlat.shared = self
+    }
+
+    @objc public static func configure(platform: String,
+                      oAuthToken: String?,
+                      userAgent: String?,
+                      anonId: String?) {
+        _ = ExPlat(configuration: ExPlatConfiguration(platform: platform,
+                                                       oAuthToken: oAuthToken,
+                                                       userAgent: userAgent,
+                                                       anonId: anonId))
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: UIApplication.didEnterBackgroundNotification, object: nil)
+        NotificationCenter.default.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+    }
+
+    /// Only refresh if the TTL has expired
+    ///
+    public func refreshIfNeeded(completion: (() -> Void)? = nil) {
+        guard ttl > 0 else {
+            completion?()
+            scheduleRefresh()
+            return
+        }
+
+        refresh(completion: completion)
+    }
+
+    /// Force the assignments to refresh
+    ///
+    public func refresh(completion: (() -> Void)? = nil) {
+        service.getAssignments { [weak self] assignments in
+            guard let `self` = self,
+                  let assignments = assignments else {
+                completion?()
+                return
+            }
+
+            let validVariations = assignments.variations.filter { $0.value != nil }
+            UserDefaults.standard.setValue(validVariations, forKey: self.assignmentsKey)
+
+            var ttlDate = Date()
+            ttlDate.addTimeInterval(TimeInterval(assignments.ttl))
+            UserDefaults.standard.setValue(ttlDate, forKey: self.ttlDateKey)
+            self.scheduleRefresh()
+
+            completion?()
+        }
+    }
+
+    public func experiment(_ name: String) -> Variation {
+        guard let assignments = UserDefaults.standard.object(forKey: assignmentsKey) as? [String: String?],
+              case let variation?? = assignments[name] else {
+            return .unknown
```

Comment:
It may be better to consider returning `.control` instead of `.unknown` so that an old client can properly fall back to the control experience if the experiment is no longer available.